File: packages/ik_rules/lib/src/pvp/arena.dart
import 'package:ik_content/ik_content.dart';

/// Citadel Plaza — the public arena.
const String arenaPlazaLocationId = 'LOC-0028';

const List<String> arenaLocationIds = <String>[arenaPlazaLocationId];

bool locationHasArena(LocationRow? location) {
  if (location == null) return false;
  return arenaLocationIds.contains(location.locationId);
}

File: src/game/pvp/arena.ts
import type { LocationRow } from '../data/types'

/** Citadel Plaza — the public arena. */
export const ARENA_PLAZA_LOCATION_ID = 'LOC-0028'

export const ARENA_LOCATION_IDS = [ARENA_PLAZA_LOCATION_ID] as const

export function locationHasArena(location: LocationRow | undefined | null): boolean {
  if (!location) return false
  return (ARENA_LOCATION_IDS as readonly string[]).includes(location['Location ID'])
}
